modules/configuration/src/test/java/org/apache/ignite/internal/configuration/notifications/ConfigurationListenerTest.java

```diff
@@ -1383,27 +1389,113 @@ void testNotificationEventForNamedConfigAfterNotifyListeners() throws Exception
+
+        config.child().listen(configListener(ctx -> events.add("child")));
+        config.child().str().listen(configListener(ctx -> events.add("child.str")));
+
+        config.children().listen(configListener(ctx -> events.add("children")));
+        config.children().listenElements(configNamedListenerOnCreate(ctx -> events.add("children.onCreate")));
+        config.children().listenElements(configNamedListenerOnUpdate(ctx -> events.add("children.onUpdate")));
+        config.children().listenElements(configNamedListenerOnRename(ctx -> events.add("children.onRename")));
+        config.children().listenElements(configNamedListenerOnDelete(ctx -> events.add("children.onDelete")));
+
+        config.children().get("0").listen(configListener(ctx -> events.add("children.0")));
+        config.children().get("0").str().listen(configListener(ctx -> events.add("children.0.str")));
+
+        config.children().any().listen(configListener(ctx -> events.add("children.any")));
+        config.children().any().str().listen(configListener(ctx -> events.add("children.any.str")));
+
+        // Polymorphic configs.
+
+        config.polyChild().listen(configListener(ctx -> events.add("polyChild")));
+        config.polyChild().commonIntVal().listen(configListener(ctx -> events.add("polyChild.int")));
+        ((StringPolyConfiguration) config.polyChild()).specificVal().listen(configListener(ctx -> events.add("polyChild.str")));
+
+        config.polyChildren().listen(configListener(ctx -> events.add("polyChildren")));
+        config.polyChildren().listenElements(configNamedListenerOnCreate(ctx -> events.add("polyChildren.onCreate")));
+        config.polyChildren().listenElements(configNamedListenerOnUpdate(ctx -> events.add("polyChildren.onUpdate")));
+        config.polyChildren().listenElements(configNamedListenerOnRename(ctx -> events.add("polyChildren.onRename")));
+        config.polyChildren().listenElements(configNamedListenerOnDelete(ctx -> events.add("polyChildren.onDelete")));
+
+        config.polyChildren().get("0").listen(configListener(ctx -> events.add("polyChildren.0")));
+        config.polyChildren().get("0").commonIntVal().listen(configListener(ctx -> events.add("polyChildren.0.int")));
+        ((StringPolyConfiguration) config.polyChildren().get("0")).specificVal()
+                .listen(configListener(ctx -> events.add("polyChildren.0.str")));
+
+        config.polyChildren().any().listen(configListener(ctx -> events.add("polyChildren.any")));
+        config.polyChildren().any().commonIntVal().listen(configListener(ctx -> events.add("polyChildren.any.int")));
+
+        Collection<CompletableFuture<?>> futs = notifyListeners(
+                null,
+                (InnerNode) config.value(),
+                (DynamicConfiguration) config,
+                0
+        );
+
+        for (CompletableFuture<?> fut : futs) {
+            fut.get(1, SECONDS);
+        }
+
+        assertEquals(
+                Set.of(
```

Comment:
Can this be a list, so that we also check the order?

Reply:
Yes, I can do it.